drivers/SmartThings/zigbee-window-treatment/src/aqara/init.lua

```diff
@@ -0,0 +1,258 @@
+          return true
+      end
+  end
+  return false
+end
+
+local read_custom_attribute = function(device, cluster_id, attribute)
+  local message = cluster_base.read_attribute(device, data_types.ClusterId(cluster_id), attribute)
+  message.body.zcl_header.frame_ctrl:set_mfg_specific()
+  message.body.zcl_header.mfg_code = data_types.validate_or_build_type(MFG_CODE, data_types.Uint16, "mfg_code")
+  return message
+end
+
+local read_pref_attribute = function(device)
+  device:send(read_custom_attribute(device, Basic.ID, PREF_ATTRIBUTE_ID))
+end
+
+local write_pref_attribute = function(device, str)
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, Basic.ID, PREF_ATTRIBUTE_ID, MFG_CODE, data_types.CharString, str))
+end
+
+local function send_open_cmd(device, component)
+  device:send_to_component(component, WindowCovering.server.commands.GoToLiftPercentage(device, 100))
+end
+
+local function send_close_cmd(device, component)
+  device:send_to_component(component, WindowCovering.server.commands.GoToLiftPercentage(device, 0))
+end
+
+local do_refresh = function(self, device)
+  device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:read(device))
+
+  read_pref_attribute(device)
+end
+
+local do_configure = function(self, device)
+  device:configure()
+
+  device:send(device_management.build_bind_request(device, WindowCovering.ID, self.environment_info.hub_zigbee_eui))
+
+  do_refresh(self, device)
+end
+
+local function device_added(driver, device)
+  local main_comp = device.profile.components["main"]
+  device:emit_component_event(main_comp,capabilities.windowShade.supportedWindowShadeCommands({"open", "close", "pause"}))
+  device:emit_component_event(main_comp,initializedstate.supportedInitializedState({"initialize"}))
+
+  write_pref_attribute(device, "\x00\x02\x00\x00\x00\x00\x00")
+  write_pref_attribute(device, "\x00\x08\x00\x00\x00\x00\x00")
+
+  device:send(Groups.server.commands.RemoveAllGroups(device))
+end
+
+local function device_info_changed(driver, device, event, args)
+  if device.preferences ~= nil then
+    if device.preferences[opencloseDirectionPreferenceId] ~= args.old_st_store.preferences[opencloseDirectionPreferenceId] then
+      if device.preferences[opencloseDirectionPreferenceId] == true then
+        write_pref_attribute(device, "\x00\x02\x00\x01\x00\x00\x00")
+      else
+        write_pref_attribute(device, "\x00\x02\x00\x00\x00\x00\x00")
+      end
+
+      device.thread:call_with_delay(2, function(d)
+        read_pref_attribute(device)
+      end)
+    end
+
+    if device.preferences[softTouchPreferenceId] ~= args.old_st_store.preferences[softTouchPreferenceId] then
+      if device.preferences[softTouchPreferenceId] == true then
+        write_pref_attribute(device, "\x00\x08\x00\x00\x00\x00\x00")
+      else
+        write_pref_attribute(device, "\x00\x08\x00\x00\x00\x01\x00")
+      end
+    end
+  end
+
+  device:send(Groups.server.commands.RemoveAllGroups(device))

```

Comment:
This should not be sent on every `infoChanged` there are lots and lots of different platform events that can cause an `infoChanged` to happen and we shouldn't just be sending a Zigbee command anytime anything about the device changes.

Reply:
@varzac What do you think about moving this code into do_configure()?, This code must be invoked whenever the sensor is rejoined by reboot.

Reply:
On any rejoin, or just when the device is factory reset, and rejoined.  If it's only on factory reset then `doConfigure` should be fine.

Reply:
Thanks Zach.
@sbkimaqara , please move this code into doConfigure() and modify the TC also.

Reply:
Updated.